x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeToCharProcessorTests.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.expression.function.scalar.datetime;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import com.carrotsearch.randomizedtesting.generators.RandomNumbers;
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+import org.junit.After;
+import org.junit.Before;
+
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.elasticsearch.xpack.ql.expression.function.scalar.FunctionTestUtils.l;
+import static org.elasticsearch.xpack.sql.expression.function.scalar.datetime.DateTimeTestUtils.dateTime;
+
+public class DateTimeToCharProcessorTests extends ESTestCase {
+
+    public static class TestCase {
+        private static final AtomicInteger COUNTER = new AtomicInteger(0);
+
+        private final int id;
+        private final BigDecimal secondsAndFractionsSinceEpoch;
+        private final String formatString;
+        private final String zoneId;
+        
+        TestCase(BigDecimal secondsAndFractionsSinceEpoch, String formatString) {
+            this.id = COUNTER.incrementAndGet();
+            this.secondsAndFractionsSinceEpoch = secondsAndFractionsSinceEpoch;
+            this.formatString = formatString;
+            this.zoneId = TestGenerator.randomFromCollection(ZoneId.getAvailableZoneIds().stream().filter(
+                z -> z.startsWith("America") || z.startsWith("Europe") || z.startsWith("Asia") || z.startsWith("Africa") 
+                    || Set.of("UTC", "GMT", "Z").contains(z)
+            ).collect(Collectors.toList()));
```

Comment:
Switching this to use offset instead of the zone names (so it will work on the CI server and with other Java versions too).